src/callbacks_step/analysis_dg1d.jl

```diff
@@ -13,8 +13,60 @@ function create_cache_analysis(analyzer,
 end
 
 
+function create_cache_analysis(analyzer, mesh::CurvedMesh,
+  equations::AbstractEquations{1}, dg::DG, cache,
+  RealT, uEltype)
+  # pre-allocate buffers
+  u_local = zeros(uEltype,
+                  nvariables(equations), nnodes(analyzer))
+  x_local = zeros(RealT,
+                  ndims(equations), nnodes(analyzer))
+  j_local = zeros(RealT, 
+                  nnodes(analyzer))
+  return (; u_local,x_local, j_local)
+end
+
+
 function calc_error_norms(func, u::AbstractArray{<:Any,3}, t, analyzer,
-                          mesh::Union{TreeMesh{1},StructuredMesh{1}}, equations, initial_condition,
+                          mesh::CurvedMesh{1, RealT}, equations, initial_condition,
+                          dg::DGSEM, cache, cache_analysis) where {RealT}
+  @unpack vandermonde, weights = analyzer
+  @unpack node_coordinates, inverse_jacobian = cache.elements
+  @unpack u_local, x_local, j_local = cache_analysis
+
+  # Set up data structures
+  l2_error   = zero(func(get_node_vars(u, equations, dg, 1, 1), equations))
+  linf_error = copy(l2_error)
+  total_volume = zero(RealT)
+
+  # Iterate over all elements for error calculations
+  for element in eachelement(dg, cache)
+    # Interpolate solution and node locations to analysis nodes
+    multiply_dimensionwise!(u_local, vandermonde, view(u,                :, :, element))
+    multiply_dimensionwise!(x_local, vandermonde, view(node_coordinates, :, :, element))
+    multiply_scalar_dimensionwise!(j_local, vandermonde, view(inverse_jacobian, :, element))
+
+    # Calculate errors at each analysis node
+    jacobian_volume = abs.(inv.(j_local))
+
+    for i in eachnode(analyzer)
+      u_exact = initial_condition(get_node_coords(x_local, equations, dg, i), t, equations)
+      diff = func(u_exact, equations) - func(get_node_vars(u_local, equations, dg, i), equations)
+      l2_error += diff.^2 * (weights[i] * jacobian_volume[i])
+      linf_error = @. max(linf_error, abs(diff))
+      total_volume += weights[i] * jacobian_volume[i]
+    end
```

Comment:
Good point - me neither! In [FLUXO](https://github.com/project-fluxo/fluxo/blob/965b73797ab27576b5a76b97d5791ba4a40f31cb/src/analyze/analyze.f90#L392-L394) and [FLEXI](https://github.com/flexi-framework/flexi/blob/913f17b9b70167333468524ac0d07ee6f60ed472/src/analyze/analyze.f90#L393-L395) they interpolate the inverse of the inverse of the Jacobian, so maybe we should just do that? In those codes is likely that somebody already gave this some thought, and since I don't really see a benefit either way, we could just follow suit.

Reply:
Yeah, that sounds more reasonable to me, too :+1:

Reply:
We couldn't find a way to invert the inverse Jacobian before interpolating without additional allocation (and without another temp array). We pushed a new version, but it allocates a bit:
```
before:
BenchmarkTools.Trial:
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     276.800 μs (0.00% GC)
  median time:      282.100 μs (0.00% GC)
  mean time:        282.631 μs (0.00% GC)
  maximum time:     402.800 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1

after:
BenchmarkTools.Trial: 
  memory estimate:  52.00 KiB
  allocs estimate:  256
  --------------
  minimum time:     286.300 μs (0.00% GC)
  median time:      292.100 μs (0.00% GC)
  mean time:        296.129 μs (0.53% GC)
  maximum time:     2.288 ms (86.54% GC)
  --------------
  samples:          10000
  evals/sample:     1
```

Reply:
That's okay for now and in accordance with our general approach.

1. Make it work
2. Make it fast

If it becomes a bottleneck, we can improve it later.